src/som/vmobjects/SClass.java

```diff
@@ -204,6 +204,16 @@ public SClass getClassCorrespondingTo(final MixinDefinitionId mixinId) {
     return cls;
   }
 
+  public SClass getClassWithMixinCorrespondingTo(final MixinDefinitionId mixinId) {
+    VM.callerNeedsToBeOptimized(
+        "This should not be on the fast path, specialization/caching needed?");
+    SClass cls = this;
+    while (cls != null && !instanceClassGroup.isBasedOn(mixinId)) {
+      cls = cls.getSuperClass();
+    }
+    return cls;
+  }
+
   @ExplodeLoop
   public SClass getClassCorrespondingTo(final int superclassIdx) {
```

Comment:
And I suppose you were just following my example? Well, looks like i changed my mind here, didn't I?
Hmmm, well, don't know. The `CorrespondingTo` seems superfluous. Maybe it isn't.
Ok, I'll leave it up to you.

Reply:
I agree that `CorrespondingTo` is not needed. I will remove. See my previous comment regarding the naming for `getClassWithMixinCorrespondingTo` and we we can go from there.